### ISSUE TITLE
VPN randomly stops working, have to reinstall script to restore access

### ISSUE THREAD

**Issue by Lawdie** [hwdsl2/setup-ipsec-vpn#104]
I have to keep reinstalling the script at random periods to restore connectivity, I can have devices connected randomly drop out while connected and no new devices are also able to connect.

I tried manually restarting various related services and none seem to restore access apart from a full reinstall

**Comment by hwdsl2** [comment]
@Lawdie Hello! Need more information about your setup. For example, which server provider and what OS? Please provide as much details as you can.

Also, when the issue appeared, were you using Libreswan 3.18 or earlier? In the new Libreswan version 3.19, which was released earlier this month, they fixed a "shunt" bug [1] [2] which can lead to the Libreswan daemon crashing. Check your version with `ipsec --version`. Try the new version and see if the issue is resolved. The VPN setup scripts have been updated to use  the new version.

[1] https://github.com/libreswan/libreswan/issues/79
[2] https://github.com/libreswan/libreswan/issues/50

**Comment by Lawdie** [comment]
Linux Libreswan 3.19 (netkey) on 4.4.0-38-generic running - Ubuntu 16.04.1 LTS

Its a Digital Ocean VPS - 512mb Ram, 1 core, 20gb SSD. The box only handles personal stuff so it is pretty much idle processing wise. When I run the VPN it is a max of 2 machines connected. 

Ill try the restart commands again the next time I run into issues (Might take upto 2 weeks before it breaks again, but ill avoid reinstalling it and update here for troubleshooting)

Cheers

**Comment by hwdsl2** [comment]
@Lawdie Thanks for the details. I'm closing this issue for now, but feel free to reply here should the issue reappear. To help troubleshooting, you can retrieve the Libreswan logs using the following command:
```
grep pluto /var/log/auth.log
```

**Comment by Lawdie** [comment]
The issue finally re appeared again, I tried connecting at. This is the last 3 lines from the log.
Feb  4 16:06:32 lawdie pluto[2541]: | setup callback for interface eth0:4500 fd 15
Feb  4 16:06:32 lawdie pluto[2541]: | setup callback for interface eth0:500 fd 14
Feb  4 16:06:32 lawdie pluto[2541]: loading secrets from "/etc/ipsec.secrets"

I tried connecting Feb 4 17:11 - there was nothing in there at all about the attempted connection

I then restarted the services and the log got updated but I was still unable to connect. Server is still in its broken state, haven't reinstall yet
